spec/spec.go

```diff
@@ -45,45 +52,29 @@ func (jf *JsonnetFile) UnmarshalJSON(data []byte) error {
 		return err
 	}
 
-	jf.Dependencies = make(map[string]Dependency)
+	jf.Dependencies = make(map[string]deps.Dependency)
 	for _, d := range s.Dependencies {
-		jf.Dependencies[d.Name] = d
+		jf.Dependencies[d.Name()] = d
 	}
+	jf.GoImportStyle = s.GoImportStyle
 	return nil
 }
 
 // MarshalJSON serializes a JsonnetFile into json of the format of a `jsonFile`
 func (jf JsonnetFile) MarshalJSON() ([]byte, error) {
 	var s jsonFile
+	s.GoImportStyle = jf.GoImportStyle
 	for _, d := range jf.Dependencies {
 		s.Dependencies = append(s.Dependencies, d)
 	}
 
 	sort.SliceStable(s.Dependencies, func(i int, j int) bool {
-		return s.Dependencies[i].Name < s.Dependencies[j].Name
+		return s.Dependencies[i].Name() < s.Dependencies[j].Name()
 	})
 
-	return json.Marshal(s)
-}
-
-type Dependency struct {
-	Name      string `json:"name"`
-	Source    Source `json:"source"`
-	Version   string `json:"version"`
-	Sum       string `json:"sum,omitempty"`
-	DepSource string `json:"-"`
-}
-
-type Source struct {
-	GitSource   *GitSource   `json:"git,omitempty"`
-	LocalSource *LocalSource `json:"local,omitempty"`
-}
-
-type GitSource struct {
-	Remote string `json:"remote"`
-	Subdir string `json:"subdir"`
-}
+	if s.Dependencies == nil {
+		s.Dependencies = make([]deps.Dependency, 0, 0)
+	}
 
-type LocalSource struct {
-	Directory string `json:"directory"`
+	return json.Marshal(s)
 }
```

Comment:
Why do we need to move these? I think it's set up in a way that one you simply import the spec package and integrate. At the same time, I don't think we have API stability guarantees here.

Reply:
I felt like parsing and working with packages (dependencies) should be separated from the specification of the config .. I would rather merge `spec` and `pkg/jsonnetfile` which seem to have the same concern. WDYT?